RetailCoder.VBE/Inspections/QuickFixes/PassParameterByReferenceQuickFix.cs

```diff
@@ -19,22 +19,19 @@ public PassParameterByReferenceQuickFix(ParserRuleContext context, QualifiedSele
 
         public override void Fix()
         {
-            var parameter = Context.GetText();
+            string pattern = "^\\s*" + Tokens.ByVal + "(\\s+)";
+            string rgxReplacement = Tokens.ByRef + "$1";
+            Regex rgx = new Regex(pattern);
 
-            var parts = parameter.Split(new char[]{' '},2);
-            if (1 != parts.GetUpperBound(0))
-            {
-                return;
-            }
-            parts[0] = parts[0].Replace(Tokens.ByVal, Tokens.ByRef);
-            var newContent = parts[0] + " " + parts[1];
+            var parameter = Context.GetText();
```

Comment:
Yeah, we actually need to clean out the code-base and consolidate\fix a lot of the code rewriting stuff.  My guess is that (outside of the indenter) most of the regular expressions are hold-overs from RD v1.x (which didn't use the same parser).  There are probably _still_ a bunch of code rewriting errors that we simply haven't found yet.